ballerina/client.bal

```diff
@@ -651,18 +652,54 @@ public isolated client class Client {
 
     }
 
-    # Get query job results
+    # Get bulk query job results
+    # 
     # + bulkJobId - Id of the bulk job
-    # + return - string[][] if successful else `error`
-    isolated remote function getQueryResult(string bulkJobId)
+    # + maxRecords - The maximum number of records to retrieve per set of results for the query
+    # + return - The resulting string[][] if successful else `error`
+    isolated remote function getQueryResult(string bulkJobId, int? maxRecords = ())
             returns string[][]|error {
-        string path = utils:prepareUrl([API_BASE_PATH, JOBS, QUERY, bulkJobId, RESULT]);
+                
+        string path = "";
+        string batchingParams = "";
+
+        if maxRecords != () {
+            lock {
+                if self.sfLocators.hasKey(bulkJobId) {
+                    string locator = self.sfLocators.get(bulkJobId);
+                    if locator is "null" {
+                        return [];
+                    } 
+                    batchingParams = string `results?maxRecords=${maxRecords}&locator=${locator}`;
+                } else {
+                    batchingParams = string `results?maxRecords=${maxRecords}`;
+                }
+            }
+            path = utils:prepareUrl([API_BASE_PATH, JOBS, QUERY, bulkJobId, batchingParams]);
+            // Max records value default, we might not know when the locator comes
+        } else {
+            lock {
+                if self.sfLocators.hasKey(bulkJobId) {
+                    string locator = self.sfLocators.get(bulkJobId);
+                    if locator is "null" {
+                        return [];
+                    } 
+                    batchingParams = string `results?locator=${locator}`;
```

Comment:
it seems we are not setting the `path` within this if statement (hence no request to be made). Is that expected?

Reply:
Nope, my mistake. 😃 Fixed it